# SNOW-512 fix http://jira.secondlife.com/browse/SNOW-512
# "Build of LLPlugin fails on 64bit linux due to non PIC code linking into the DSO (2.0 code base)"
Index: indra/llcommon/CMakeLists.txt
===================================================================
--- indra/llcommon/CMakeLists.txt	(revision 3198)
+++ indra/llcommon/CMakeLists.txt	(working copy)
@@ -250,6 +250,13 @@
 
 if(LLCOMMON_LINK_SHARED)
     add_library (llcommon SHARED ${llcommon_SOURCE_FILES})
+	if(NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
+      if(WINDOWS)
+        add_definitions(/FIXED:NO)
+      else(WINDOWS) # not windows therefore gcc LINUX and DARWIN
+        add_definitions(-fPIC)
+      endif(WINDOWS)
+	endif(NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
     ll_stage_sharedlib(llcommon)
 else(LLCOMMON_LINK_SHARED)
     add_library (llcommon ${llcommon_SOURCE_FILES})
--- indra/media_plugins/webkit/CMakeLists.txt	(revision 3198)
+++ indra/media_plugins/webkit/CMakeLists.txt	(working copy)
@@ -34,6 +38,14 @@
     media_plugin_webkit.cpp
     )
 
+if(NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
+	if(WINDOWS)
+		add_definitions(/FIXED:NO)
+	else(WINDOWS) # not windows therefore gcc LINUX and DARWIN
+		add_definitions(-fPIC)
+	endif(WINDOWS)
+endif (NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
+
 add_library(media_plugin_webkit
     SHARED
     ${media_plugin_webkit_SOURCE_FILES}
Index: indra/media_plugins/base/CMakeLists.txt
===================================================================
--- indra/media_plugins/base/CMakeLists.txt	(revision 3198)
+++ indra/media_plugins/base/CMakeLists.txt	(working copy)
@@ -25,6 +29,14 @@
 
 ### media_plugin_base
 
+if(NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
+	if(WINDOWS)
+		add_definitions(/FIXED:NO)
+	else(WINDOWS) # not windows therefore gcc LINUX and DARWIN
+		add_definitions(-fPIC)
+	endif(WINDOWS)
+endif (NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
+
 set(media_plugin_base_SOURCE_FILES
     media_plugin_base.cpp
 )
Index: indra/media_plugins/gstreamer010/CMakeLists.txt
===================================================================
--- indra/media_plugins/gstreamer010/CMakeLists.txt	(revision 3198)
+++ indra/media_plugins/gstreamer010/CMakeLists.txt	(working copy)
@@ -30,6 +34,14 @@
 
 ### media_plugin_gstreamer010
 
+if(NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
+	if(WINDOWS)
+		add_definitions(/FIXED:NO)
+	else(WINDOWS) # not windows therefore gcc LINUX and DARWIN
+		add_definitions(-fPIC)
+	endif(WINDOWS)
+endif (NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
+
 set(media_plugin_gstreamer010_SOURCE_FILES
     media_plugin_gstreamer010.cpp
     llmediaimplgstreamer_syms.cpp
Index: indra/media_plugins/example/CMakeLists.txt
===================================================================
--- indra/media_plugins/example/CMakeLists.txt	(revision 3198)
+++ indra/media_plugins/example/CMakeLists.txt	(working copy)
@@ -29,6 +33,14 @@
 
 ### media_plugin_example
 
+if(NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
+	if(WINDOWS)
+		add_definitions(/FIXED:NO)
+	else(WINDOWS) # not windows therefore gcc LINUX and DARWIN
+		add_definitions(-fPIC)
+	endif(WINDOWS)
+endif (NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
+
 set(media_plugin_example_SOURCE_FILES
     media_plugin_example.cpp
     )
Index: indra/llplugin/CMakeLists.txt
===================================================================
--- indra/llplugin/CMakeLists.txt	(revision 3198)
+++ indra/llplugin/CMakeLists.txt	(working copy)
@@ -48,6 +52,14 @@
 set_source_files_properties(${llplugin_HEADER_FILES}
                             PROPERTIES HEADER_FILE_ONLY TRUE)
 
+if(NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
+	if(WINDOWS)
+		add_definitions(/FIXED:NO)
+	else(WINDOWS) # not windows therefore gcc LINUX and DARWIN
+		add_definitions(-fPIC)
+	endif(WINDOWS)
+endif (NOT CMAKE_SIZEOF_VOID_P MATCHES 4)
+
 list(APPEND llplugin_SOURCE_FILES ${llplugin_HEADER_FILES})
 
 add_library (llplugin ${llplugin_SOURCE_FILES})

